plugins/woocommerce/src/Admin/Features/ProductBlockEditor/Init.php

```diff
@@ -78,21 +72,9 @@ public function enqueue_scripts() {
 		if ( ! PageController::is_admin_or_embed_page() ) {
 			return;
 		}
-		$post_type_object     = get_post_type_object( 'product' );
-		$block_editor_context = new WP_Block_Editor_Context( array( 'name' => self::EDITOR_CONTEXT_NAME ) );
-		$template_registry    = wc_get_container()->get( BlockTemplateRegistry::class );
 
-		$editor_settings = array();
-		if ( ! empty( $post_type_object->template ) ) {
-			$editor_settings['template']     = $post_type_object->template;
-			$editor_settings['templateLock'] = ! empty( $post_type_object->template_lock ) ? $post_type_object->template_lock : false;
```

Comment:
This line is still necessary within your new `get_product_editor_settings()` as the block picker started showing up in the variation template, I am not sure why not in the simple product template. Maybe we do have template lock hardcoded for that template within the JS somewhere?
<img width="840" alt="Screenshot 2023-11-14 at 11 50 29 AM" src="https://github.com/woocommerce/woocommerce/assets/2240960/a4514c2e-166d-438d-8309-8d138da6e5d4">

Reply:
Good catch. I'm going to fix this in the JS, as we always want `templateLock` set to `all`. We shouldn't be relying on the PHP to set this properly.